src/main/java/Deadline.java

```diff
@@ -0,0 +1,39 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task of type Deadline.
+ * A Deadline is represented by a name in the form of a String,
+ * and the date by which the task should be completed, in the form
+ * of a LocalDate.
+ */
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    /**
+     * Creates a Deadline with the description and date by which the task should be completed.
+     * @param name Description of the Deadline.
+     * @param by The date by which the task should be completed.
+     */
+    public Deadline(String name, LocalDate by) {
+        super(name);
+        this.by = by;
+    }
+
+    /**
+     * Parses the date in the format of "MMM dd yyyy".
+     * @param date Date to be parsed.
+     * @return A String representing the date in the new format.
+     */
+    public String parseDate(LocalDate date) {
+        return date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
```

Comment:
"Avoid magic numbers" https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers

Might want to use a named constant for this?